apps/internal/base/base.go

```diff
@@ -312,6 +314,12 @@ func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilen
 	// ignore cached access tokens when given claims
 	if silent.Claims == "" {
 		ar, err = AuthResultFromStorage(storageTokenResponse)
+		if err == nil && authParams.AuthenticationScheme != nil {
```

Comment:
We need to make sure that the token cache schema is the same across MSALs. The simplest is to serialize the cache to a plaintext file and observe it's structure. Note that each access token has a "cache key" and a "cache payload".

For POP tokens, we don't want the `kid` to be part of the access token cache key. This is to prevent the token cache becoming too large, as keys can rotate frequently. 

The changes needed are: 

- add the "schema name" (i.e. "pop") in the key. You'll have to add it to `IAuthenticationScheme` as well. 
- add the `kid` to the payload payload
- ensure `token_type` is in the payload as well

```json
"AccessToken": {
    "-login.windows.net-accesstoken_with_authscheme-16dab2ba-145d-4b1b-8569-bf4b9aed4dc8-72f988bf-86f1-41af-91ab-2d7cd011db47-https://vault.azure.net/.default-pop": {
      "environment": "login.windows.net",
      "client_id": "16dab2ba-145d-4b1b-8569-bf4b9aed4dc8",
      "secret": "actual_token",
      "credential_type": "AccessToken",
      "realm": "72f988bf-86f1-41af-91ab-2d7cd011db47",
      "target": "https://vault.azure.net/.default",
      "cached_at": "1688149697",
      "expires_on": "1688236096",
      "extended_expires_on": "1688236096",
      "kid": "TJtUe-3Wa8X6cOxKzmeTxjHDO1-gXBKtazWh0xx3SA0",    # new!
      "token_type": "pop",   # not Bearer!
      "refresh_on": "1688192896",
      "ext_expires_on": "1688236096"
    }
```